synapseclient/core/upload/multipart_upload_async.py

```diff
@@ -609,6 +641,18 @@ def _put_part_with_retry(
 
                 else:
                     raise
+
+        self.record_span_event(
```

Comment:
Might be helpful to walk through and define a span, and then events in a span, and then spans in that event and how all of that impacts the dashboarding in signoz / SQL queries in clickhouse